File: src/#1/classes.js
// @flow

class Foo {
  a: number;
  constructor(a: number) {
    this.a = a;
  }
}

class Bar {
  a: number;
  constructor(a: number) {
    this.a = a;
  }
}

const foo: Foo = new Foo(0);
const bar: Foo = new Bar(0);

File: src/1/classes.cpp
class Foo {
  public:
    int a;

  Foo(int a): a(a) {}
};

class Bar {
  public:
    int a;

  Bar(int a): a(a) {}
};

int main() {
  Foo* foo = new Foo(0);
  Bar* bar = new Foo(0);
}